File: data/multilang_comparison/program_go/src/func39.go
package main
import "math/rand"
var _ = rand.Intn
func Func39(vars *ArrayParam, loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array56 *Array
   if pCounter > 0 {
       array56 = vars.Data[pCounter-1]
       pCounter--
       array56.RefC++
   } else {
       array56 = &Array{
           Size: 143,
           RefC: 1,
           Id: 56,
           Data: make([]uint32, 143),
       }
   }
   var params0 ArrayParam
   params0.Size = 1
   params0.Data = make([]*Array, params0.Size)
   params0.Data[0] = array56
   var array57 *Array = Func41(&params0, loopsFactor);
   params0.Data = nil
   params0.Size = 0
   array56.RefC--
   if array56.RefC == 0 {
       array56.Data = nil
       array56 = nil
   }
   return array57;
}

File: data/multilang_comparison/program_cpp_vector/src/func39.cpp
#include "program_cpp_vector.hpp" 
Array* func39(Array_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   Array* array56;
   if (pCounter > 0) {
      array56 = vars->data.at(--pCounter);
      array56->refC++;
      DEBUG_COPY(array56->id);
   } else {
      array56 = new Array();
      array56->size = 143;
      array56->refC = 1;
      array56->id = 56;
      array56->data = std::vector<unsigned int>(array56->size, 0);
      DEBUG_NEW(array56->id);
   }
   Array_param params0;
   params0.size = 1;
   std::vector<Array*> data_params0(1);
   params0.data = data_params0;
   params0.data.at(0) = array56;
   Array* array57 = func41(&params0, loopsFactor);
   DEBUG_RETURN(array57->id);
   if(params0.size > 0) {
   	 params0.data.clear();
   }
   array56->refC--;
   if(array56->refC == 0) {
      array56->data.clear();
      delete array56;
      DEBUG_FREE(array56->id);
   }
   return array57;
}

